utils/vote-state/src/lib.rs

```diff
@@ -471,20 +537,180 @@ mod tests {
             epoch_credits: test_epoch_credits,
             last_timestamp: BlockTimestamp::default(),
         }));
-        let mut ser_current = bincode::serialize(&vote_state_current).unwrap();
+        let mut ser_1_16_0 = bincode::serialize(&vote_state_1_16_0).unwrap();
         let account_info = AccountInfo::new(
             &key,
             false,
             false,
             &mut lamports,
-            ser_current.as_mut_slice(),
+            ser_1_16_0.as_mut_slice(),
             &owner,
             false,
             0,
         );
         let epoch_credits_result =
             VoteStateVersions::deserialize_epoch_credits(&account_info).unwrap();
         assert!(epoch_credits_result == vec![(70, 9, 6), (321, 20, 4)]);
+
+        // Test empty and non-empty variants of variable length fields
+
+        let test_epoch_credits: Vec<(Epoch, u64, u64)> = vec![(70, 9, 6), (321, 20, 4)];
+        let vote_state_current = VoteStateVersions::Current(Box::new(crate::VoteState {
+            node_pubkey: Pubkey::new_unique(),
+            authorized_withdrawer: Pubkey::new_unique(),
+            inflation_rewards_collector: Pubkey::new_unique(),
+            block_revenue_collector: Pubkey::new_unique(),
+            inflation_rewards_commission_bps: 99,
+            block_revenue_commission_bps: 99,
+            pending_delegator_rewards: 0,
+            bls_pubkey_compressed: Some(BLSPubkey { bytes: [0; 48] }),
+            votes: VecDeque::new(),
+            root_slot: Some(1),
+            authorized_voters: AuthorizedVoters::default(),
+            epoch_credits: test_epoch_credits.clone(),
+            last_timestamp: BlockTimestamp::default(),
+        }));
+        let mut ser_current = bincode::serialize(&vote_state_current).unwrap();
+        let account_current = AccountInfo::new(
+            &key,
+            false,
+            false,
+            &mut lamports,
+            ser_current.as_mut_slice(),
+            &owner,
+            false,
+            0,
+        );
+        let epoch_credits_result =
+            VoteStateVersions::deserialize_epoch_credits(&account_current).unwrap();
+        assert!(epoch_credits_result == test_epoch_credits);
```

Comment:
nit: I think `assert_eq!` is better, when a test fails, assert_eq! gives us a helpful diff showing both values which makes debugging much faster.

```suggestion
        assert_eq!(epoch_credits_result, test_epoch_credits);
```